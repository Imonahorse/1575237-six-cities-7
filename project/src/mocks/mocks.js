import {AuthorizationStatus} from '../const.js';

const fakeOffers = [
  {
    city: {
      name: 'Hamburg',
      location: {
        latitude: 53.550341,
        longitude: 10.000654,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/6.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/13.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/19.jpg',
      'https://7.react.pages.academy/static/hotel/16.jpg',
      'https://7.react.pages.academy/static/hotel/11.jpg',
      'https://7.react.pages.academy/static/hotel/14.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
    ],
    title: 'House in countryside',
    isFavorite: false,
    isPremium: false,
    rating: 2.2,
    type: 'room',
    bedrooms: 1,
    maxAdults: 1,
    price: 177,
    goods: [
      'Breakfast',
      'Laptop friendly workspace',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.',
    location: {
      latitude: 53.558341000000006,
      longitude: 10.001654,
      zoom: 16,
    },
    id: 1,
  },
  {
    city: {
      name: 'Amsterdam',
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/2.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/11.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/14.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/16.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/10.jpg',
      'https://7.react.pages.academy/static/hotel/19.jpg',
    ],
    title: 'The house among olive ',
    isFavorite: false,
    isPremium: false,
    rating: 3.7,
    type: 'room',
    bedrooms: 1,
    maxAdults: 3,
    price: 260,
    goods: [
      'Laptop friendly workspace',
      'Washer',
      'Breakfast',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    location: {
      latitude: 52.36854,
      longitude: 4.887976,
      zoom: 16,
    },
    id: 2,
  },
  {
    city: {
      name: 'Cologne',
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/4.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/13.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/1.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/11.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/10.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
    ],
    title: 'House in countryside',
    isFavorite: false,
    isPremium: true,
    rating: 4.3,
    type: 'house',
    bedrooms: 2,
    maxAdults: 3,
    price: 738,
    goods: [
      'Laptop friendly workspace',
      'Breakfast',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    location: {
      latitude: 50.916361,
      longitude: 6.944974,
      zoom: 16,
    },
    id: 3,
  },
  {
    city: {
      name: 'Amsterdam',
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/6.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/13.jpg',
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/1.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/11.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/14.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
    ],
    title: 'Wood and stone place',
    isFavorite: false,
    isPremium: false,
    rating: 5,
    type: 'room',
    bedrooms: 1,
    maxAdults: 2,
    price: 155,
    goods: [
      'Breakfast',
      'Laptop friendly workspace',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.',
    location: {
      latitude: 52.35754,
      longitude: 4.9179759999999995,
      zoom: 16,
    },
    id: 4,
  },
  {
    city: {
      name: 'Cologne',
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/16.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/14.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/1.jpg',
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/20.jpg',
    ],
    title: 'Canal View Prinsengracht',
    isFavorite: false,
    isPremium: false,
    rating: 3.6,
    type: 'hotel',
    bedrooms: 2,
    maxAdults: 7,
    price: 191,
    goods: [
      'Breakfast',
      'Laptop friendly workspace',
      'Washer',
      'Air conditioning',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.',
    location: {
      latitude: 50.949361,
      longitude: 6.976974,
      zoom: 16,
    },
    id: 5,
  },
  {
    city: {
      name: 'Paris',
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/8.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/16.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/13.jpg',
      'https://7.react.pages.academy/static/hotel/19.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/10.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/20.jpg',
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/14.jpg',
    ],
    title: 'The Pondhouse - A Magical Place',
    isFavorite: true,
    isPremium: false,
    rating: 2.6,
    type: 'house',
    bedrooms: 3,
    maxAdults: 6,
    price: 443,
    goods: [
      'Air conditioning',
      'Breakfast',
      'Laptop friendly workspace',
      'Baby seat',
      'Washer',
      'Towels',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.',
    location: {
      latitude: 48.85761,
      longitude: 2.358499,
      zoom: 16,
    },
    id: 6,
  },
  {
    city: {
      name: 'Dusseldorf',
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/11.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/14.jpg',
      'https://7.react.pages.academy/static/hotel/16.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/20.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/1.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/10.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
    ],
    title: 'Nice, cozy, warm big bed apartment',
    isFavorite: false,
    isPremium: false,
    rating: 3.6,
    type: 'room',
    bedrooms: 1,
    maxAdults: 2,
    price: 125,
    goods: [
      'Air conditioning',
      'Laptop friendly workspace',
      'Breakfast',
      'Washer',
      'Towels',
      'Baby seat',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.',
    location: {
      latitude: 51.241402,
      longitude: 6.782314,
      zoom: 16,
    },
    id: 7,
  },
  {
    city: {
      name: 'Brussels',
      location: {
        latitude: 50.846557,
        longitude: 4.351697,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/9.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/11.jpg',
      'https://7.react.pages.academy/static/hotel/20.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/19.jpg',
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/5.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
    ],
    title: 'Wood and stone place',
    isFavorite: true,
    isPremium: false,
    rating: 2.5,
    type: 'hotel',
    bedrooms: 2,
    maxAdults: 3,
    price: 135,
    goods: [
      'Towels',
      'Breakfast',
      'Baby seat',
      'Fridge',
      'Air conditioning',
      'Laptop friendly workspace',
      'Washer',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.',
    location: {
      latitude: 50.842557,
      longitude: 4.363696999999999,
      zoom: 16,
    },
    id: 8,
  },
  {
    city: {
      name: 'Hamburg',
      location: {
        latitude: 53.550341,
        longitude: 10.000654,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/2.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/10.jpg',
      'https://7.react.pages.academy/static/hotel/16.jpg',
      'https://7.react.pages.academy/static/hotel/1.jpg',
      'https://7.react.pages.academy/static/hotel/13.jpg',
      'https://7.react.pages.academy/static/hotel/14.jpg',
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
    ],
    title: 'Waterfront with extraordinary view',
    isFavorite: false,
    isPremium: true,
    rating: 2.5,
    type: 'house',
    bedrooms: 3,
    maxAdults: 5,
    price: 547,
    goods: [
      'Washer',
      'Breakfast',
      'Air conditioning',
      'Baby seat',
      'Laptop friendly workspace',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'This is a place for dreamers to reset, reflect, and create. Designed with a \'slow\' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.',
    location: {
      latitude: 53.565341000000004,
      longitude: 9.978654,
      zoom: 16,
    },
    id: 9,
  },
  {
    city: {
      name: 'Brussels',
      location: {
        latitude: 50.846557,
        longitude: 4.351697,
        zoom: 13,
      },
    },
    previewImage: 'https://7.react.pages.academy/static/hotel/5.jpg',
    images: [
      'https://7.react.pages.academy/static/hotel/2.jpg',
      'https://7.react.pages.academy/static/hotel/12.jpg',
      'https://7.react.pages.academy/static/hotel/16.jpg',
      'https://7.react.pages.academy/static/hotel/9.jpg',
      'https://7.react.pages.academy/static/hotel/4.jpg',
      'https://7.react.pages.academy/static/hotel/11.jpg',
      'https://7.react.pages.academy/static/hotel/6.jpg',
      'https://7.react.pages.academy/static/hotel/15.jpg',
      'https://7.react.pages.academy/static/hotel/18.jpg',
      'https://7.react.pages.academy/static/hotel/8.jpg',
      'https://7.react.pages.academy/static/hotel/17.jpg',
      'https://7.react.pages.academy/static/hotel/7.jpg',
      'https://7.react.pages.academy/static/hotel/3.jpg',
      'https://7.react.pages.academy/static/hotel/19.jpg',
    ],
    title: 'Perfectly located Castro',
    isFavorite: false,
    isPremium: false,
    rating: 4.9,
    type: 'house',
    bedrooms: 4,
    maxAdults: 10,
    price: 243,
    goods: [
      'Breakfast',
      'Laptop friendly workspace',
    ],
    host: {
      id: 25,
      name: 'Angelina',
      isPro: true,
      avatarUrl: 'img/avatar-angelina.jpg',
    },
    description: 'I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.',
    location: {
      latitude: 50.849557,
      longitude: 4.374696999999999,
      zoom: 16,
    },
    id: 10,
  },
];
const fakeComments = [
  {
    id: 1,
    user: {
      id: 17,
      isPro: false,
      name: 'Emely',
      avatarUrl: 'https://7.react.pages.academy/static/avatar/8.jpg',
    },
    rating: 4,
    comment: 'I stayed here for one night and it was an unpleasant experience.',
    date: '2021-07-10T11:09:43.784Z',
  },
  {
    id: 2,
    user: {
      id: 1,
      isPro: false,
      name: 'Alex',
      avatarUrl: 'https://7.react.pages.academy/static/avatar/6.jpg',
    },
    rating: 4,
    comment: 'Comment text.',
    date: '2021-07-23T18:45:52.327Z',
  },
  {
    id: 3,
    user: {
      id: 1,
      isPro: false,
      name: 'awsfdfd',
      avatarUrl: 'https://7.react.pages.academy/static/avatar/6.jpg',
    },
    rating: 1,
    comment: 'I stayed here for one night and it was an unpleasant experience.I stayed here for one night and it was an unpleasant experience.I stayed here for one night and it was an unpleasant experience.',
    date: '2021-07-23T18:45:59.095Z',
  },
];
const fakeStore = {
  USER: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    logoutStatus: { isSuccess: false, isLoading: false, isError: false},
    loginStatus: { isSuccess: false, isLoading: false, isError: false},
    favoriteStatus: { isSuccess: false, isLoading: false, isError: false},
    setFavoriteStatus: { isSuccess: false, isLoading: false, isError: false},
  },
  APP: {
    city: 'Paris',
  },
  DATA: {
    offers: fakeOffers,
    offer: fakeOffers[0],
    offerStatus: { isSuccess: false, isLoading: false, isError: false},
    comments: fakeComments,
    nearPlacesOffersStatus: { isSuccess: false, isLoading: false, isError: false},
    nearPlacesOffers: fakeOffers,
    commentsStatus: { isSuccess: false, isLoading: false, isError: false},
    offersStatus: { isSuccess: false, isLoading: false, isError: false},
  },
};

export {fakeOffers, fakeComments, fakeStore};
