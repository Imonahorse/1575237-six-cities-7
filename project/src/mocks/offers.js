import {nanoid} from 'nanoid';
import {CITIES} from '../const.js';

const AVATAR_URL = 'https://i.pravatar.cc/128';
const APARTMENT_URL = 'http://lorempixel.com/260/200/city/';
const OFFERS_COUNT = 10;
const TITLES = [
  'St. Francis Psychiatric Hospital',
  'Sell garage',
  'Freaks ahead, factories await you',
  'Iam blue dabudidabudai',
];
const TYPES = [
  'Apartment',
  'Kennel',
  'Garbage tent',
  'Under the bridge',
  'Place in the cemetery',
];
const FEATURES = [
  'Wi-Fi',
  'Washing machine',
  'Towels',
  'Heating',
  'Coffee machine',
  'Baby seat',
  'Kitchen',
  'Dishwasher',
  'Cabe TV',
  'Fridge',
];
const NAMES = [
  'Ashot',
  'Jamshut',
  'Gogi',
  'Maga',
  'Rocky',
  'Bulvinkl',
];
const DESCRIPTIONS = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et doloremagna aliqua.',
  'Platea dictumst vestibulum rhoncus est pellentesque elit. Sit amet tellus cras adipiscing enim eu turpis egestas.',
  'Ornare arcu dui vivamus arcu felis bibendum ut tristique et. Magna sit amet purus gravida quis.',
  'Lorem ipsum dolor sit amet consectetur',
  'Nibh ipsum consequat nisl vel pretium lectus quam id leo.',
  'Tellus molestie nunc non blandit massa enim nec dui nunc.',
];

const getRandomInteger = (lower, upper) => lower + Math.floor(Math.random() * (upper - lower + 1));
const getRandomArrayElement = (array) => array[getRandomInteger(0, array.length - 1)];
const getRandomArray = (array) => array.filter(() => Math.random() > 0.3);
const getRandomImage = () => ({
  src: APARTMENT_URL + getRandomInteger(1, 10),
  alt: getRandomArrayElement(FEATURES),
});
const getImageArray = () => new Array(getRandomInteger(1, 6)).fill('').map(() => getRandomImage());

const createOffer = () => ({
  city: getRandomArrayElement(CITIES),
  description: getRandomArray(DESCRIPTIONS),
  isPremium: Math.random() > 0.3,
  isFavorite: Math.random() > 0.8,
  title: getRandomArrayElement(TITLES),
  rating: getRandomInteger(1, 5),
  type: getRandomArrayElement(TYPES),
  bedroomsCount: getRandomInteger(0, 5),
  maxAdults: getRandomInteger(0, 10),
  price: getRandomInteger(100, 1000),
  features: getRandomArray(FEATURES),
  host: {
    name: getRandomArrayElement(NAMES),
    isPro: Math.random() > 0.3,
    avatarUrl: `${AVATAR_URL}?rnd=${Math.random()}`,
    id: nanoid(),
  },
  id: nanoid(),
  images: getImageArray(),
  previewImage: getRandomImage(),
});

const createOffers = () => new Array(OFFERS_COUNT).fill('').map(() => createOffer());

export {createOffers};
