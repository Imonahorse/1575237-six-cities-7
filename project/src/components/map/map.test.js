import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import Map from './map.jsx';
import {render} from '@testing-library/react';
import {fakeOffers} from '../../mocks/mocks.js';

const offers = fakeOffers;
const {id} = offers[0];

describe('Component: Map', () => {
  it('should render map correctly', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <Map
          cityState={offers[0].city.name}
          cityOffers={offers}
          activeCardId={id}
        />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
