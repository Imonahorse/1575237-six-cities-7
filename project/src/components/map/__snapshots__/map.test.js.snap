// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Map should render map correctly 1`] = `
<DocumentFragment>
  <section
    class="map cities__map leaflet-container leaflet-grab leaflet-touch-drag"
    data-testid="map"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://b.tile.thunderforest.com/spinal-map/12/2161/1323.png?apikey=b42a4fe28722439084924941798385fc"
              style="width: 256px; height: 256px; left: -181px; top: -203px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: 23px; top: -35px; z-index: -35;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -14872px; top: 5719px; z-index: 5719;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -8880px; top: 12535px; z-index: 12535;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -14784px; top: 5771px; z-index: 5771;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -8787px; top: 12383px; z-index: 12383;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -22239px; top: 21844px; z-index: 21844;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -9354px; top: 11028px; z-index: 11028;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -16399px; top: 12876px; z-index: 12876;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -44px; top: -70px; z-index: -70;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -39px; width: 27px; height: 39px; left: -16367px; top: 12844px; z-index: 12844;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="http://www.thunderforest.com/"
          >
            Thunderforest
          </a>
          , © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
